vispy/visuals/image.py

```diff
@@ -400,8 +400,14 @@ def _build_texture(self):
             # deal with clim on CPU b/c of texture depth limits :(
             # can eventually do this by simulating 32-bit float... maybe
             clim = self._clim
+            # deal with masked_array
+            if np.ma.is_masked(data):
+                mask = data.mask
+                data = np.array(data)
+                data[mask] = np.nan
+            is_num = np.isfinite(data)
```

Comment:
Can we move this inside the `if` statement below so that it is only computed if it is needed?

Reply:
I would say no because non-numerical values can happen in non-masked arrays

Reply:
Sure, but I'm saying you create `is_num` above the if statement and then don't use it until you are *inside* the if statement. It should be created inside the if statement to avoid needlessly creating it (when clim is not `'auto'`).

Reply:
Oh yes, sorry, I didn't read your message correctly. Yes, indeed, I would be better inside the `if isinstance(clim, string_types) and clim == 'auto':`